pkg/apis/eventing/v1alpha1/subscription_types.go

```diff
@@ -0,0 +1,200 @@
+	//
+	// You can specify only the following fields of the ObjectReference:
+	//   - Kind
+	//   - APIVersion
+	//   - Name
+	// This field is immutable
+	From *corev1.ObjectReference `json:"from,omitempty"`
+
+	// Call is reference to (optional) function for processing events.
+	// Events from the From channel will be delivered here and replies
+	// are sent to a channel as specified by the Result.
+	// +optional
+	Call *Callable `json:"call,omitempty"`
+
+	// Result specifies (optionally) how to handle events returned from
+	// the Call target.
+	// +optional
+	Result *ResultStrategy `json:"to,omitempty"`
+}
+
+// Callable specifies the reference to an object that's expected to
+// provide the resolved target of the action. Currently we inspect
+// the objects Status and see if there's a predefined Status field
+// that we will then use to dispatch events to be processed by the target.
+// Currently must resolve to a k8s service or Istio virtual service. Note that
+// in the future we should try to utilize subresources (/resolve ?) to
+// make this cleaner, but CRDs do not support subresources yet, so we need
+// to rely on a specified Status field today. By relying on this behaviour
+// we can utilize a dynamic client instead of having to understand all
+// kinds of different types of objects. As long as they adhere to this
+// particular contract, they can be used as a Target.
+// This ensures that we can support external targets and for ease of use
+// we also allow for an URI to be specified.
+// There of course is also a requirement for the resolved Callable to
+// behave properly at the data plane level.
+type Callable struct {
+	// Only one of these can be specified
+
+	// Reference to an object that will be used to find the target
+	// endpoint.
+	// For example, this could be a reference to a Route resource
+	// or a Knative Service resource.
+	// TODO: Specify the required fields the target object must
+	// have in the status.
+	// You can specify only the following fields of the ObjectReference:
+	//   - Kind
+	//   - APIVersion
+	//   - Name
+	// +optional
+	Target *corev1.ObjectReference `json:"target,omitempty"`
+
+	// Reference to a 'known' endpoint where no resolving is done.
+	// http://k8s-service for example
+	// http://myexternalhandler.example.com/foo/bar
+	// +optional
+	TargetURI *string `json:"targetURI,omitempty"`
+}
+
+type ResultStrategy struct {
+	// This object must fulfill the Sinkable contract.
+	//
+	// TODO: Specify the required fields the target object must
+	// have in the status.
+	// You can specify only the following fields of the ObjectReference:
+	//   - Kind
+	//   - APIVersion
+	//   - Name
+	// +optional
+	Target *corev1.ObjectReference `json:"target,omitempty"`
+}
+
+type SubscriptionConditionType string
+
+const (
+	// Dispatching means the subscription is actively listening for incoming events on its channel and dispatching them.
+	SubscriptionDispatching SubscriptionConditionType = "Dispatching"
```

Comment:
Is the meaning of `Dispatching` different from the meaning of `Ready` in serving?

Reply:
I haven't touched the status, it's old cruft. Wanted to get Spec right first.